concoction-lib/src/main/java/info/mmpa/concoction/scan/model/dynamic/entry/Condition.java

```diff
@@ -0,0 +1,27 @@
+package info.mmpa.concoction.scan.model.dynamic.entry;
+
+import info.mmpa.concoction.scan.dynamic.CallStackFrame;
+
+import javax.annotation.Nonnull;
+
+// TODO: Sub-types
+//  - Any (wildcard, useful where any execution of a 'MethodLocation' is flag-worthy)
+//  - Value matching for (Parameter N), (Field $NAME/$DESC)
+//    - Primitive math matching
+//    - Object null/not-null
+//    - String TextMatchMode matching
+//      - Also for char[] and other similar types (StringBuilder, StringBuffer, non-string CharSequence)
+//    - byte[] matching helpers
```

Comment:
TODO: The condition sub-types need to be implemented still + the serializer/deserializer _(and this class annotated to use them)_